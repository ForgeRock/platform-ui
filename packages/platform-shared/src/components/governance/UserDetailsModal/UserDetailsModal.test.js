/**
 * Copyright (c) 2023 ForgeRock. All rights reserved.
 *
 * This software may be modified and distributed under the terms
 * of the MIT license. See the LICENSE file for details.
 */

import { shallowMount } from '@vue/test-utils';
import GovernanceUserDetailsModal from './index';

describe('GovernanceUserDetailsModal', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(GovernanceUserDetailsModal, {
      global: {
        mocks: {
          $t: (t) => t,
        },
      },
      props: {
        user: {
          givenName: 'firstname',
          sn: 'sntest',
          username: 'test',
        },
        userDetails: {
          userAccounts: {},
          userEntitlements: {},
          userRoles: {},
        },
        userEntitlements: {
          result: [{
            application: {
              id: 'test-id',
              name: 'test-name',
              icon: 'test-url',
            },
          }],
        },
      },
    });
  });

  it('component should load correclty', () => {
    expect(wrapper.vm.userFullName).toBe('common.userFullName');
  });
});
