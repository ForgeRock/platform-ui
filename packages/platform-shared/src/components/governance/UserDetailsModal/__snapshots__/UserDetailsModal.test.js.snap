// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GovernanceUserDetailsModal component should load correclty: console-warn-snapshot 1`] = `
[
  [
    "[Vue warn]: (deprecation RENDER_FUNCTION) Vue 3's render function API has changed. You can opt-in to the new API with:

  configureCompat({ RENDER_FUNCTION: false })

  (This can also be done per-component via the "compatConfig" option.)
  Details: https://v3-migration.vuejs.org/breaking-changes/render-function-api.html",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation GLOBAL_EXTEND) (2)",
    "
",
    " at <GovernanceUserDetailsModal",
    "user=",
    {
      "givenName": "firstname",
      "sn": "sntest",
      "username": "test",
    },
    "userDetails=",
    {
      "userAccounts": {},
      "userEntitlements": {},
      "userRoles": {},
    },
    "userEntitlements=",
    {
      "result": [
        {
          "application": {
            "icon": "test-url",
            "id": "test-id",
            "name": "test-name",
          },
        },
      ],
    },
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "visible" with v-bind value \`false\` will render visible="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BModal",
    "ok-title="common.done"",
    "body-class="p-0"",
    "content-class="border-0"",
    " ...",
    ">",
    "
",
    " at <GovernanceUserDetailsModal",
    "user=",
    {
      "givenName": "firstname",
      "sn": "sntest",
      "username": "test",
    },
    "userDetails=",
    {
      "userAccounts": {},
      "userEntitlements": {},
      "userRoles": {},
    },
    "userEntitlements=",
    {
      "result": [
        {
          "application": {
            "icon": "test-url",
            "id": "test-id",
            "name": "test-name",
          },
        },
      ],
    },
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "busy" with v-bind value \`false\` will render busy="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BModal",
    "ok-title="common.done"",
    "body-class="p-0"",
    "content-class="border-0"",
    " ...",
    ">",
    "
",
    " at <GovernanceUserDetailsModal",
    "user=",
    {
      "givenName": "firstname",
      "sn": "sntest",
      "username": "test",
    },
    "userDetails=",
    {
      "userAccounts": {},
      "userEntitlements": {},
      "userRoles": {},
    },
    "userEntitlements=",
    {
      "result": [
        {
          "application": {
            "icon": "test-url",
            "id": "test-id",
            "name": "test-name",
          },
        },
      ],
    },
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "cancelDisabled" with v-bind value \`false\` will render cancelDisabled="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BModal",
    "ok-title="common.done"",
    "body-class="p-0"",
    "content-class="border-0"",
    " ...",
    ">",
    "
",
    " at <GovernanceUserDetailsModal",
    "user=",
    {
      "givenName": "firstname",
      "sn": "sntest",
      "username": "test",
    },
    "userDetails=",
    {
      "userAccounts": {},
      "userEntitlements": {},
      "userRoles": {},
    },
    "userEntitlements=",
    {
      "result": [
        {
          "application": {
            "icon": "test-url",
            "id": "test-id",
            "name": "test-name",
          },
        },
      ],
    },
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "centered" with v-bind value \`false\` will render centered="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BModal",
    "ok-title="common.done"",
    "body-class="p-0"",
    "content-class="border-0"",
    " ...",
    ">",
    "
",
    " at <GovernanceUserDetailsModal",
    "user=",
    {
      "givenName": "firstname",
      "sn": "sntest",
      "username": "test",
    },
    "userDetails=",
    {
      "userAccounts": {},
      "userEntitlements": {},
      "userRoles": {},
    },
    "userEntitlements=",
    {
      "result": [
        {
          "application": {
            "icon": "test-url",
            "id": "test-id",
            "name": "test-name",
          },
        },
      ],
    },
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "hideBackdrop" with v-bind value \`false\` will render hideBackdrop="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BModal",
    "ok-title="common.done"",
    "body-class="p-0"",
    "content-class="border-0"",
    " ...",
    ">",
    "
",
    " at <GovernanceUserDetailsModal",
    "user=",
    {
      "givenName": "firstname",
      "sn": "sntest",
      "username": "test",
    },
    "userDetails=",
    {
      "userAccounts": {},
      "userEntitlements": {},
      "userRoles": {},
    },
    "userEntitlements=",
    {
      "result": [
        {
          "application": {
            "icon": "test-url",
            "id": "test-id",
            "name": "test-name",
          },
        },
      ],
    },
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "hideFooter" with v-bind value \`false\` will render hideFooter="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BModal",
    "ok-title="common.done"",
    "body-class="p-0"",
    "content-class="border-0"",
    " ...",
    ">",
    "
",
    " at <GovernanceUserDetailsModal",
    "user=",
    {
      "givenName": "firstname",
      "sn": "sntest",
      "username": "test",
    },
    "userDetails=",
    {
      "userAccounts": {},
      "userEntitlements": {},
      "userRoles": {},
    },
    "userEntitlements=",
    {
      "result": [
        {
          "application": {
            "icon": "test-url",
            "id": "test-id",
            "name": "test-name",
          },
        },
      ],
    },
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "hideHeader" with v-bind value \`false\` will render hideHeader="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BModal",
    "ok-title="common.done"",
    "body-class="p-0"",
    "content-class="border-0"",
    " ...",
    ">",
    "
",
    " at <GovernanceUserDetailsModal",
    "user=",
    {
      "givenName": "firstname",
      "sn": "sntest",
      "username": "test",
    },
    "userDetails=",
    {
      "userAccounts": {},
      "userEntitlements": {},
      "userRoles": {},
    },
    "userEntitlements=",
    {
      "result": [
        {
          "application": {
            "icon": "test-url",
            "id": "test-id",
            "name": "test-name",
          },
        },
      ],
    },
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "hideHeaderClose" with v-bind value \`false\` will render hideHeaderClose="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BModal",
    "ok-title="common.done"",
    "body-class="p-0"",
    "content-class="border-0"",
    " ...",
    ">",
    "
",
    " at <GovernanceUserDetailsModal",
    "user=",
    {
      "givenName": "firstname",
      "sn": "sntest",
      "username": "test",
    },
    "userDetails=",
    {
      "userAccounts": {},
      "userEntitlements": {},
      "userRoles": {},
    },
    "userEntitlements=",
    {
      "result": [
        {
          "application": {
            "icon": "test-url",
            "id": "test-id",
            "name": "test-name",
          },
        },
      ],
    },
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "lazy" with v-bind value \`false\` will render lazy="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BModal",
    "ok-title="common.done"",
    "body-class="p-0"",
    "content-class="border-0"",
    " ...",
    ">",
    "
",
    " at <GovernanceUserDetailsModal",
    "user=",
    {
      "givenName": "firstname",
      "sn": "sntest",
      "username": "test",
    },
    "userDetails=",
    {
      "userAccounts": {},
      "userEntitlements": {},
      "userRoles": {},
    },
    "userEntitlements=",
    {
      "result": [
        {
          "application": {
            "icon": "test-url",
            "id": "test-id",
            "name": "test-name",
          },
        },
      ],
    },
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "noEnforceFocus" with v-bind value \`false\` will render noEnforceFocus="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BModal",
    "ok-title="common.done"",
    "body-class="p-0"",
    "content-class="border-0"",
    " ...",
    ">",
    "
",
    " at <GovernanceUserDetailsModal",
    "user=",
    {
      "givenName": "firstname",
      "sn": "sntest",
      "username": "test",
    },
    "userDetails=",
    {
      "userAccounts": {},
      "userEntitlements": {},
      "userRoles": {},
    },
    "userEntitlements=",
    {
      "result": [
        {
          "application": {
            "icon": "test-url",
            "id": "test-id",
            "name": "test-name",
          },
        },
      ],
    },
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "noFade" with v-bind value \`false\` will render noFade="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BModal",
    "ok-title="common.done"",
    "body-class="p-0"",
    "content-class="border-0"",
    " ...",
    ">",
    "
",
    " at <GovernanceUserDetailsModal",
    "user=",
    {
      "givenName": "firstname",
      "sn": "sntest",
      "username": "test",
    },
    "userDetails=",
    {
      "userAccounts": {},
      "userEntitlements": {},
      "userRoles": {},
    },
    "userEntitlements=",
    {
      "result": [
        {
          "application": {
            "icon": "test-url",
            "id": "test-id",
            "name": "test-name",
          },
        },
      ],
    },
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "noStacking" with v-bind value \`false\` will render noStacking="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BModal",
    "ok-title="common.done"",
    "body-class="p-0"",
    "content-class="border-0"",
    " ...",
    ">",
    "
",
    " at <GovernanceUserDetailsModal",
    "user=",
    {
      "givenName": "firstname",
      "sn": "sntest",
      "username": "test",
    },
    "userDetails=",
    {
      "userAccounts": {},
      "userEntitlements": {},
      "userRoles": {},
    },
    "userEntitlements=",
    {
      "result": [
        {
          "application": {
            "icon": "test-url",
            "id": "test-id",
            "name": "test-name",
          },
        },
      ],
    },
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "okDisabled" with v-bind value \`false\` will render okDisabled="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BModal",
    "ok-title="common.done"",
    "body-class="p-0"",
    "content-class="border-0"",
    " ...",
    ">",
    "
",
    " at <GovernanceUserDetailsModal",
    "user=",
    {
      "givenName": "firstname",
      "sn": "sntest",
      "username": "test",
    },
    "userDetails=",
    {
      "userAccounts": {},
      "userEntitlements": {},
      "userRoles": {},
    },
    "userEntitlements=",
    {
      "result": [
        {
          "application": {
            "icon": "test-url",
            "id": "test-id",
            "name": "test-name",
          },
        },
      ],
    },
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "static" with v-bind value \`false\` will render static="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BModal",
    "ok-title="common.done"",
    "body-class="p-0"",
    "content-class="border-0"",
    " ...",
    ">",
    "
",
    " at <GovernanceUserDetailsModal",
    "user=",
    {
      "givenName": "firstname",
      "sn": "sntest",
      "username": "test",
    },
    "userDetails=",
    {
      "userAccounts": {},
      "userEntitlements": {},
      "userRoles": {},
    },
    "userEntitlements=",
    {
      "result": [
        {
          "application": {
            "icon": "test-url",
            "id": "test-id",
            "name": "test-name",
          },
        },
      ],
    },
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
  [
    "[Vue warn]: (deprecation ATTR_FALSE_VALUE) Attribute "titleSrOnly" with v-bind value \`false\` will render titleSrOnly="false" instead of removing it in Vue 3. To remove the attribute, use \`null\` or \`undefined\` instead. If the usage is intended, you can disable the compat behavior and suppress this warning with:

  configureCompat({ ATTR_FALSE_VALUE: false })

  Details: https://v3-migration.vuejs.org/breaking-changes/attribute-coercion.html",
    "
",
    " at <BModal",
    "ok-title="common.done"",
    "body-class="p-0"",
    "content-class="border-0"",
    " ...",
    ">",
    "
",
    " at <GovernanceUserDetailsModal",
    "user=",
    {
      "givenName": "firstname",
      "sn": "sntest",
      "username": "test",
    },
    "userDetails=",
    {
      "userAccounts": {},
      "userEntitlements": {},
      "userRoles": {},
    },
    "userEntitlements=",
    {
      "result": [
        {
          "application": {
            "icon": "test-url",
            "id": "test-id",
            "name": "test-name",
          },
        },
      ],
    },
    " ...",
    ">",
    "
",
    " at <VTUROOT>",
  ],
]
`;
